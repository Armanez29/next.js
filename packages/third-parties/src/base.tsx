import React from 'react'

export default function Base({
  content,
  height = null,
  width = null,
  children,
}: {
  content?: string
  height?: number | null
  width?: number | null
  children?: React.ReactElement | React.ReactElement[]
}) {
  return (
    <>
      {/* insert script children */}
      {children}
      {/* insert content */}
      {content && (
        <div
          style={{
            height: `${height}px` || 'auto',
            width: `${width}px` || 'auto',
          }}
          dangerouslySetInnerHTML={{ __html: content }}
        />
      )}
    </>
  )
}
