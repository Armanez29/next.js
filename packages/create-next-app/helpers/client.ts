/* eslint-disable import/no-extraneous-dependencies */
import got from 'got'
import { HttpsProxyAgent } from 'hpagent'

function getAgent() {
  if (process.env.https_proxy) {
    return {
      https: new HttpsProxyAgent({
        proxy: process.env.https_proxy,
        rejectUnauthorized: false,
      }),
    }
  }
}

export default got.extend({ agent: getAgent() })
