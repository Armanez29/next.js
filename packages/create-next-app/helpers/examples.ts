/* eslint-disable import/no-extraneous-dependencies */
import tar from 'tar'
import { Stream, Readable } from 'stream'
import { promisify } from 'util'
import { join } from 'path'
import { tmpdir } from 'os'
import { createWriteStream, promises as fs } from 'fs'

const pipeline = promisify(Stream.pipeline)

export type RepoInfo = {
  username: string
  name: string
  branch: string
  filePath: string
}

export async function isUrlOk(url: string): Promise<boolean> {
  try {
    const res = await fetch(url, { method: 'HEAD' })
    if (res.status !== 200) {
      return false
    }
    return true
  } catch {
    return false
  }
}

export async function getRepoInfo(
  url: URL,
  examplePath?: string
): Promise<RepoInfo | undefined> {
  const [, username, name, t, _branch, ...file] = url.pathname.split('/')
  const filePath = examplePath ? examplePath.replace(/^\//, '') : file.join('/')

  if (
    // Support repos whose entire purpose is to be a Next.js example, e.g.
    // https://github.com/:username/:my-cool-nextjs-example-repo-name.
    t === undefined ||
    // Support GitHub URL that ends with a trailing slash, e.g.
    // https://github.com/:username/:my-cool-nextjs-example-repo-name/
    // In this case "t" will be an empty string while the next part "_branch" will be undefined
    (t === '' && _branch === undefined)
  ) {
    try {
      const infoResponse = await fetch(
        `https://api.github.com/repos/${username}/${name}`
      )
      if (infoResponse.status !== 200) {
        return
      }

      const info = await infoResponse.json()
      return { username, name, branch: info['default_branch'], filePath }
    } catch {
      return
    }
  }

  // If examplePath is available, the branch name takes the entire path
  const branch = examplePath
    ? `${_branch}/${file.join('/')}`.replace(new RegExp(`/${filePath}|/$`), '')
    : _branch

  if (username && name && branch && t === 'tree') {
    return { username, name, branch, filePath }
  }
}

export function hasRepo({
  username,
  name,
  branch,
  filePath,
}: RepoInfo): Promise<boolean> {
  const contentsUrl = `https://api.github.com/repos/${username}/${name}/contents`
  const packagePath = `${filePath ? `/${filePath}` : ''}/package.json`

  return isUrlOk(contentsUrl + packagePath + `?ref=${branch}`)
}

export function existsInRepo(nameOrUrl: string): Promise<boolean> {
  try {
    const url = new URL(nameOrUrl)
    return isUrlOk(url.href)
  } catch {
    return isUrlOk(
      `https://api.github.com/repos/vercel/next.js/contents/examples/${encodeURIComponent(
        nameOrUrl
      )}`
    )
  }
}

async function downloadTar(url: string) {
  const tempFile = join(tmpdir(), `next.js-cna-example.temp-${Date.now()}`)

  const resp = await fetch(url)

  if (!resp.body) {
    throw new Error(`Failed to download: ${url}`)
  }

  await pipeline(
    Readable.fromWeb(resp.body as import('stream/web').ReadableStream),
    createWriteStream(tempFile)
  )

  return tempFile
}

export async function downloadAndExtractRepo(
  root: string,
  { username, name, branch, filePath }: RepoInfo
) {
  const tempFile = await downloadTar(
    `https://codeload.github.com/${username}/${name}/tar.gz/${branch}`
  )

  await tar.x({
    file: tempFile,
    cwd: root,
    strip: filePath ? filePath.split('/').length + 1 : 1,
    filter: (p) =>
      p.startsWith(
        `${name}-${branch.replace(/\//g, '-')}${
          filePath ? `/${filePath}/` : '/'
        }`
      ),
  })

  await fs.unlink(tempFile)
}

export async function downloadAndExtractExample(root: string, name: string) {
  if (name === '__internal-testing-retry') {
    throw new Error('This is an internal example for testing the CLI.')
  }

  const tempFile = await downloadTar(
    'https://codeload.github.com/vercel/next.js/tar.gz/canary'
  )

  await tar.x({
    file: tempFile,
    cwd: root,
    strip: 2 + name.split('/').length,
    filter: (p) => p.includes(`next.js-canary/examples/${name}/`),
  })

  await fs.unlink(tempFile)
}
