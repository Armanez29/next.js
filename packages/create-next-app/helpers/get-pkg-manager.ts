export type PackageManager = 'npm' | 'pnpm' | 'yarn'

export function getPkgManager(): PackageManager {
  const userAgent = process.env.npm_config_user_agent

  if (userAgent && userAgent.startsWith('yarn')) {
    return 'yarn'
  }
  
  if (userAgent && userAgent.startsWith('pnpm')) {
    return 'pnpm'
  } 
  
  return 'npm'
}
