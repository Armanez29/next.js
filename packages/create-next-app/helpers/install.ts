/* eslint-disable import/no-extraneous-dependencies */
import chalk from 'chalk'
import spawn from 'cross-spawn'

interface InstallArgs {
  /**
   * Indicate whether to install packages using Yarn.
   */
  useYarn: boolean
  /**
   * Indicate whether there is an active Internet connection.
   */
  isOnline: boolean
  /**
   * Indicate whether the given dependencies are devDependencies.
   */
  devDependencies?: boolean
}

export function install(
  root: string,
  dependencies: string[] | null,
  { useYarn, isOnline, devDependencies }: InstallArgs
): Promise<void> {
  return new Promise((resolve, reject) => {
    let command: string
    let args: string[] = []
    if (useYarn) {
      /**
       * Installing with Yarn.
       */
      command = 'yarnpkg'
      if (dependencies && dependencies.length) {
        args = ['add', '--exact']
        if (!isOnline) args.push('--offline')
        if (devDependencies) args.push('-D')
        args.push(...dependencies)
        args.push('--cwd', root)
      } else {
        /**
         * Run `yarn install` if no deps.
         */
        args = ['install']
      }
      if (!isOnline) {
        console.log(chalk.yellow('You appear to be offline.'))
        console.log(chalk.yellow('Falling back to the local Yarn cache.'))
        console.log()
      }
    } else {
      /**
       * Installing with NPM.
       */
      command = 'npm'
      const npmFlags = ['--logLevel', 'error']
      if (dependencies && dependencies.length) {
        args = ['install', '--save-exact']
        args.push(devDependencies ? '--save-dev' : '--save')
        args.push(...dependencies)
      } else {
        /**
         * Run `npm install` if no deps.
         */
        args = ['install']
      }
      args.push(...npmFlags)
    }

    const child = spawn(command, args, {
      stdio: 'inherit',
      env: { ...process.env, ADBLOCK: '1', DISABLE_OPENCOLLECTIVE: '1' },
    })
    child.on('close', (code) => {
      if (code !== 0) {
        reject({ command: `${command} ${args.join(' ')}` })
        return
      }
      resolve()
    })
  })
}
