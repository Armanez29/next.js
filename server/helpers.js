import { join } from 'path'
import fs from 'mz/fs'
import glob from 'glob-promise'

export async function getParamRoutes (base) {
  let files = await glob(join(base, '**'))

  const result = await Promise.all(files.map(f => isFile(f)))
  files = result.reduce((_files, isFile, i) => {
    return isFile ? [..._files, files[i]] : _files
  })

  files = files.filter(f => f.indexOf('}') > 0)
  files = files.map(f => f.replace(base, ''))
  return files
}

export async function isFile (p) {
  let stat
  try {
    stat = await fs.stat(p)
  } catch (err) {
    if (err.code === 'ENOENT') return false
    throw err
  }
  return stat.isFile() || stat.isFIFO()
}

export function toParts (route) {
  return route.replace(/^\/|\.js$/g, '').split('/')
}

export function compareRoute (parts, segments) {
  // parts:    ['user', '{id}', 'action']
  // segments: ['user', '1234', 'action']
  if (parts.length === segments.length) {
    return parts.map((p, i) => p.startsWith('{') || segments[i] === parts[i]).every(ok => ok)
  }
  return false
}
