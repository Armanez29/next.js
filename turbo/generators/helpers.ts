import type { PlopTypes } from '@turbo/gen'

export function toFileName(str: string) {
  return str.toLowerCase().replace(/ /g, '-')
}

export function init(plop: PlopTypes.NodePlopAPI): void {
  plop.setHelper('toFileName', toFileName)
}
