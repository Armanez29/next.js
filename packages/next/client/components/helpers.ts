import type { CacheNode } from '../../shared/lib/app-router-context'
import type { FlightRouterState } from '../../server/app-render'
import { fetchServerResponse } from './app-router.client'

export const createOptimisticTree = (
  existingState: any,
  url: URL,
  // /dashboard/integrations ['', 'dashboard', 'integrations', 'page']
  segments: string[],
  existingFlightRouterState: FlightRouterState | null,
  childNodes: CacheNode['childNodes'] | null,
  isFirstSegment: boolean,
  parentFetchingData: boolean,
  initialMap = new Map()
): {
  childNodes: CacheNode['childNodes']
  tree: FlightRouterState
} => {
  const [existingSegment, existingParallelRoutes] =
    existingFlightRouterState || [null, {}]
  const segment = segments[0]
  const isLastSegment = segments.length === 1

  let parallelRoutes: FlightRouterState[1] = {}
  if (existingSegment === segment) {
    parallelRoutes = existingParallelRoutes
  }

  if (childNodes) {
    for (const [key, value] of childNodes) {
      initialMap.set(key, {
        subTreeData: value.subTreeData,
        childNodes: new Map(),
      })
    }
  }

  let startedFetch = false
  if (!isFirstSegment && !parentFetchingData) {
    // Seed the cache with missing item
    if (childNodes && !childNodes.has(segment)) {
      startedFetch = true
      console.log('KICKING OFF DATA FETCH IN PUSH', {
        segment,
        url,
      })

      initialMap.set(segment, {
        data: fetchServerResponse(url, existingState.tree),
        subTreeData: null,
        childNodes: new Map(),
      })
    }
  }

  let childTree
  if (!isLastSegment) {
    const passedChildNodes = isFirstSegment
      ? childNodes
      : childNodes && childNodes.has(segment)
      ? childNodes.get(segment)!.childNodes
      : null
    const childItem = createOptimisticTree(
      existingState,
      url,
      segments.slice(1),
      parallelRoutes ? parallelRoutes.children : null,
      passedChildNodes,
      false,
      parentFetchingData || startedFetch,
      isFirstSegment ? initialMap : initialMap.get(segment)!.childNodes
    )

    childTree = childItem.tree
  }

  const result: FlightRouterState = [
    segment,
    {
      ...parallelRoutes,
      ...(childTree ? { children: childTree } : {}),
    },
  ]

  // Add url into the tree
  if (isFirstSegment) {
    result[2] = url.pathname + url.search
  }

  return {
    childNodes: initialMap,
    tree: result,
  }
}
