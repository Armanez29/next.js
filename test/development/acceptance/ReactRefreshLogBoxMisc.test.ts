import { sandbox } from './helpers'
import { createNext } from 'e2e-utils'
import { NextInstance } from 'test/lib/next-modes/base'

describe('ReactRefreshLogBox', () => {
  let next: NextInstance

  beforeAll(async () => {
    next = await createNext({
      files: {},
      skipStart: true,
    })
  })
  afterAll(() => next.destroy())

  test('<Link> with multiple children', async () => {
    const { session, cleanup } = await sandbox(next)

    await session.patch(
      'index.js',
      `
        import Link from 'next/link'

        export default function Index() {
          return (
            <Link href="/">
              <p>One</p>
              <p>Two</p>
            </Link>
          )
        }
      `
    )

    expect(await session.hasRedbox(true)).toBe(true)
    expect(await session.getRedboxDescription()).toMatchInlineSnapshot(
      `"Error: Multiple children were passed to <Link> with \`href\` of \`/\` but only one child is supported https://nextjs.org/docs/messages/link-multiple-children"`
    )
    expect(
      await session.evaluate(
        () =>
          (
            document
              .querySelector('body > nextjs-portal')
              .shadowRoot.querySelector(
                '#nextjs__container_errors_desc a:nth-of-type(1)'
              ) as any
          ).href
      )
    ).toMatch('https://nextjs.org/docs/messages/link-multiple-children')

    await cleanup()
  })

  test('<Link> component props errors', async () => {
    const { session, cleanup } = await sandbox(next)

    await session.patch(
      'index.js',
      `
        import Link from 'next/link'

        export default function Hello() {
          return <Link />
        }
      `
    )

    expect(await session.hasRedbox(true)).toBe(true)
    expect(await session.getRedboxDescription()).toMatchInlineSnapshot(
      `"Error: Failed prop type: The prop \`href\` expects a \`string\` or \`object\` in \`<Link>\`, but got \`undefined\` instead."`
    )

    await session.patch(
      'index.js',
      `
        import Link from 'next/link'

        export default function Hello() {
          return <Link href="/">Abc</Link>
        }
      `
    )
    expect(await session.hasRedbox()).toBe(false)

    await session.patch(
      'index.js',
      `
        import Link from 'next/link'

        export default function Hello() {
          return (
            <Link
              href="/"
              as="/"
              replace={false}
              scroll={false}
              shallow={false}
              passHref={false}
              prefetch={false}
            >
              Abc
            </Link>
          )
        }
      `
    )
    expect(await session.hasRedbox()).toBe(false)

    await session.patch(
      'index.js',
      `
        import Link from 'next/link'

        export default function Hello() {
          return (
            <Link
              href="/"
              as="/"
              replace={true}
              scroll={true}
              shallow={true}
              passHref={true}
              prefetch={true}
            >
              Abc
            </Link>
          )
        }
      `
    )
    expect(await session.hasRedbox()).toBe(false)

    await session.patch(
      'index.js',
      `
        import Link from 'next/link'

        export default function Hello() {
          return (
            <Link
              href="/"
              as="/"
              replace={undefined}
              scroll={undefined}
              shallow={undefined}
              passHref={undefined}
              prefetch={undefined}
            >
              Abc
            </Link>
          )
        }
      `
    )
    expect(await session.hasRedbox()).toBe(false)

    await session.patch(
      'index.js',
      `
        import Link from 'next/link'

        export default function Hello() {
          return (
            <Link
              href="/"
              as="/"
              replace={undefined}
              scroll={'oops'}
              shallow={undefined}
              passHref={undefined}
              prefetch={undefined}
            >
              Abc
            </Link>
          )
        }
      `
    )
    expect(await session.hasRedbox(true)).toBe(true)
    expect(await session.getRedboxDescription()).toMatchSnapshot()

    await session.patch(
      'index.js',
      `
        import Link from 'next/link'

        export default function Hello() {
          return (
            <Link
              href={false}
              as="/"
              replace={undefined}
              scroll={'oops'}
              shallow={undefined}
              passHref={undefined}
              prefetch={undefined}
            >
              Abc
            </Link>
          )
        }
      `
    )
    expect(await session.hasRedbox(true)).toBe(true)
    expect(await session.getRedboxDescription()).toMatchSnapshot()

    await cleanup()
  })

  test('_app top level error shows logbox', async () => {
    const { session, cleanup } = await sandbox(
      next,
      new Map([
        [
          'pages/_app.js',
          `
            throw new Error("test");
            function MyApp({ Component, pageProps }) {
              return <Component {...pageProps} />;
            }
            export default MyApp
          `,
        ],
      ])
    )
    expect(await session.hasRedbox(true)).toBe(true)
    expect(await session.getRedboxSource()).toMatchSnapshot()

    await session.patch(
      'pages/_app.js',
      `
        function MyApp({ Component, pageProps }) {
          return <Component {...pageProps} />;
        }
        export default MyApp
      `
    )
    expect(await session.hasRedbox()).toBe(false)
    await cleanup()
  })

  test('_document top level error shows logbox', async () => {
    const { session, cleanup } = await sandbox(
      next,
      new Map([
        [
          'pages/_document.js',
          `
            import Document, { Html, Head, Main, NextScript } from 'next/document'

            throw new Error("test");

            class MyDocument extends Document {
              static async getInitialProps(ctx) {
                const initialProps = await Document.getInitialProps(ctx)
                return { ...initialProps }
              }

              render() {
                return (
                  <Html>
                    <Head />
                    <body>
                      <Main />
                      <NextScript />
                    </body>
                  </Html>
                )
              }
            }

            export default MyDocument
          `,
        ],
      ])
    )
    expect(await session.hasRedbox(true)).toBe(true)
    expect(await session.getRedboxSource()).toMatchSnapshot()

    await session.patch(
      'pages/_document.js',
      `
        import Document, { Html, Head, Main, NextScript } from 'next/document'

        class MyDocument extends Document {
          static async getInitialProps(ctx) {
            const initialProps = await Document.getInitialProps(ctx)
            return { ...initialProps }
          }

          render() {
            return (
              <Html>
                <Head />
                <body>
                  <Main />
                  <NextScript />
                </body>
              </Html>
            )
          }
        }

        export default MyDocument
      `
    )
    expect(await session.hasRedbox()).toBe(false)
    await cleanup()
  })

  // TODO: investigate why this test is so flakey
  test.skip('server-side only compilation errors', async () => {
    const { session, cleanup } = await sandbox(next)

    await session.patch(
      'pages/index.js',
      `
        import myLibrary from 'my-non-existent-library'
        export async function getStaticProps() {
          return {
            props: {
              result: myLibrary()
            }
          }
        }
        export default function Hello(props) {
          return <h1>{props.result}</h1>
        }
      `
    )

    expect(await session.hasRedbox(true)).toBe(true)
    await cleanup()
  })

  test('empty _app shows logbox', async () => {
    const { session, cleanup } = await sandbox(
      next,
      new Map([
        [
          'pages/_app.js',
          `
            
          `,
        ],
      ])
    )
    expect(await session.hasRedbox(true)).toBe(true)
    expect(await session.getRedboxDescription()).toMatchInlineSnapshot(
      `"Error: The default export is not a React Component in page: \\"/_app\\""`
    )

    await session.patch(
      'pages/_app.js',
      `
        function MyApp({ Component, pageProps }) {
          return <Component {...pageProps} />;
        }
        export default MyApp
      `
    )
    expect(await session.hasRedbox()).toBe(false)
    await cleanup()
  })

  test('empty _document shows logbox', async () => {
    const { session, cleanup } = await sandbox(
      next,
      new Map([
        [
          'pages/_document.js',
          `
            
          `,
        ],
      ])
    )
    expect(await session.hasRedbox(true)).toBe(true)
    expect(await session.getRedboxDescription()).toMatchInlineSnapshot(
      `"Error: The default export is not a React Component in page: \\"/_document\\""`
    )

    await session.patch(
      'pages/_document.js',
      `
        import Document, { Html, Head, Main, NextScript } from 'next/document'

        class MyDocument extends Document {
          static async getInitialProps(ctx) {
            const initialProps = await Document.getInitialProps(ctx)
            return { ...initialProps }
          }

          render() {
            return (
              <Html>
                <Head />
                <body>
                  <Main />
                  <NextScript />
                </body>
              </Html>
            )
          }
        }

        export default MyDocument
      `
    )
    expect(await session.hasRedbox()).toBe(false)
    await cleanup()
  })

  test('_app syntax error shows logbox', async () => {
    const { session, cleanup } = await sandbox(
      next,
      new Map([
        [
          'pages/_app.js',
          `
            function MyApp({ Component, pageProps }) {
              return <<Component {...pageProps} />;
            }
            export default MyApp
          `,
        ],
      ])
    )
    expect(await session.hasRedbox(true)).toBe(true)
    expect(await session.getRedboxSource()).toMatchSnapshot()

    await session.patch(
      'pages/_app.js',
      `
        function MyApp({ Component, pageProps }) {
          return <Component {...pageProps} />;
        }
        export default MyApp
      `
    )
    expect(await session.hasRedbox()).toBe(false)
    await cleanup()
  })

  test('_document syntax error shows logbox', async () => {
    const { session, cleanup } = await sandbox(
      next,
      new Map([
        [
          'pages/_document.js',
          `
            import Document, { Html, Head, Main, NextScript } from 'next/document'

            class MyDocument extends Document {{
              static async getInitialProps(ctx) {
                const initialProps = await Document.getInitialProps(ctx)
                return { ...initialProps }
              }

              render() {
                return (
                  <Html>
                    <Head />
                    <body>
                      <Main />
                      <NextScript />
                    </body>
                  </Html>
                )
              }
            }

            export default MyDocument
          `,
        ],
      ])
    )
    expect(await session.hasRedbox(true)).toBe(true)
    expect(await session.getRedboxSource()).toMatchSnapshot()

    await session.patch(
      'pages/_document.js',
      `
        import Document, { Html, Head, Main, NextScript } from 'next/document'

        class MyDocument extends Document {
          static async getInitialProps(ctx) {
            const initialProps = await Document.getInitialProps(ctx)
            return { ...initialProps }
          }

          render() {
            return (
              <Html>
                <Head />
                <body>
                  <Main />
                  <NextScript />
                </body>
              </Html>
            )
          }
        }

        export default MyDocument
      `
    )
    expect(await session.hasRedbox()).toBe(false)
    await cleanup()
  })

  // Module trace is only available with webpack 5
  if (!process.env.NEXT_PRIVATE_TEST_WEBPACK4_MODE) {
    test('Node.js builtins', async () => {
      const { session, cleanup } = await sandbox(
        next,
        new Map([
          [
            'node_modules/my-package/index.js',
            `
            const dns = require('dns')
            module.exports = dns
          `,
          ],
          [
            'node_modules/my-package/package.json',
            `
            {
              "name": "my-package",
              "version": "0.0.1"
            }
          `,
          ],
        ])
      )

      await session.patch(
        'index.js',
        `
        import pkg from 'my-package'

        export default function Hello() {
          return (pkg ? <h1>Package loaded</h1> : <h1>Package did not load</h1>)
        }
      `
      )
      expect(await session.hasRedbox(true)).toBe(true)
      expect(await session.getRedboxSource()).toMatchSnapshot()

      await cleanup()
    })
  }
})
