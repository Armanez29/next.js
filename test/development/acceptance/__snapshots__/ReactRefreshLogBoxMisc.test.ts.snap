// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactRefreshLogBox <Link> component props errors 2`] = `
"Error: Failed prop type: The prop \`scroll\` expects a \`boolean\` in \`<Link>\`, but got \`string\` instead.
Open your browser's console to view the Component stack trace."
`;

exports[`ReactRefreshLogBox <Link> component props errors 3`] = `
"Error: Failed prop type: The prop \`href\` expects a \`string\` or \`object\` in \`<Link>\`, but got \`boolean\` instead.
Open your browser's console to view the Component stack trace."
`;

exports[`ReactRefreshLogBox _app syntax error shows logbox 1`] = `
"./pages/_app.js:3:22
Syntax error: Unexpected token

  1 |
  2 |             function MyApp({ Component, pageProps }) {
> 3 |               return <<Component {...pageProps} />;
    |                       ^
  4 |             }
  5 |             export default MyApp
  6 |"
`;

exports[`ReactRefreshLogBox _app top level error shows logbox 1`] = `
"pages/_app.js (2:18) @ eval

  1 | 
> 2 |             throw new Error(\\"test\\");
    |                  ^
  3 |             function MyApp({ Component, pageProps }) {
  4 |               return <Component {...pageProps} />;
  5 |             }"
`;

exports[`ReactRefreshLogBox _document syntax error shows logbox 1`] = `
"./pages/_document.js:4:47
Syntax error: Unexpected token

  2 |             import Document, { Html, Head, Main, NextScript } from 'next/document'
  3 |
> 4 |             class MyDocument extends Document {{
    |                                                ^
  5 |               static async getInitialProps(ctx) {
  6 |                 const initialProps = await Document.getInitialProps(ctx)
  7 |                 return { ...initialProps }"
`;

exports[`ReactRefreshLogBox _document top level error shows logbox 1`] = `
"pages/_document.js (4:18) @ eval

  2 |             import Document, { Html, Head, Main, NextScript } from 'next/document'
  3 | 
> 4 |             throw new Error(\\"test\\");
    |                  ^
  5 | 
  6 |             class MyDocument extends Document {
  7 |               static async getInitialProps(ctx) {"
`;

exports[`ReactRefreshLogBox Node.js builtins 1`] = `
"./node_modules/my-package/index.js:2:0
Module not found: Can't resolve 'dns'

Import trace for requested module:
./index.js
./pages/index.js

https://nextjs.org/docs/messages/module-not-found"
`;

exports[`ReactRefreshLogBox Node.js builtins 2`] = `
"./node_modules/my-package/index.js:2:0
Module not found: Can't resolve 'dns'

Import trace for requested module:
./index.js
./pages/index.js

https://nextjs.org/docs/messages/module-not-found"
`;

exports[`ReactRefreshLogBox Node.js builtins 3`] = `
"./node_modules/my-package/index.js:2:0
Module not found: Can't resolve 'dns'

Import trace for requested module:
./index.js
./pages/index.js

https://nextjs.org/docs/messages/module-not-found"
`;
